src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -195,18 +197,41 @@ public FeatureManagementKeyValueAdapter(FeatureFilterTracing featureFilterTracin
 
                 string telemetryPath = $"{featureFlagPath}:{FeatureManagementConstants.Telemetry}";
 
-                if (telemetry.Enabled)
-                {
-                    keyValues.Add(new KeyValuePair<string, string>($"{telemetryPath}:{FeatureManagementConstants.Enabled}", telemetry.Enabled.ToString()));
-                }
-
                 if (telemetry.Metadata != null)
                 {
                     foreach (KeyValuePair<string, string> kvp in telemetry.Metadata)
                     {
                         keyValues.Add(new KeyValuePair<string, string>($"{telemetryPath}:{FeatureManagementConstants.Metadata}:{kvp.Key}", kvp.Value));
                     }
                 }
+
+                if (telemetry.Enabled)
+                {
+                    byte[] featureFlagIdHash;
+
+                    using (HashAlgorithm hashAlgorithm = SHA256.Create())
+                    {
+                        featureFlagIdHash = hashAlgorithm.ComputeHash(Encoding.UTF8.GetBytes($"{setting.Key}\n{setting.Label}"));
```

Comment:
I don't recall what the plan was- but if label is null- are we omitting the `\n` and null? I think currently this would give the string: 

```text
Key
null
```

Reply:
It should evaluate to "Key\n", but I'm not sure if we clarified whether we want to keep the newline character as part of the value even if there's a null label? I think it's fine as it is but just wanted to call it out.

Reply:
Do we differentiate null vs. empty label?

Reply:
I assumed we wouldn't, when I tried on portal I couldn't pass an empty string, and we also treat whitespace/null as the same for label filters in calls to `Select`.